pkg/test/framework/resource/context.go

```diff
@@ -20,20 +20,37 @@ import (
 	"istio.io/istio/pkg/test/util/yml"
 )
 
-type ConfigOptions struct {
-	NoCleanup bool
-	Wait      bool
+type ApplyOptions struct {
+	Cleanup CleanupStrategy
+	Wait    bool
 }
 
-type ConfigOption func(o *ConfigOptions)
+type ApplyOption func(o *ApplyOptions)
 
-// NoCleanup does not delete the applied Config once it goes out of scope.
-var NoCleanup ConfigOption = func(o *ConfigOptions) {
-	o.NoCleanup = true
+// CleanupStrategy enumerates the options for configuration cleanup behavior.
+type CleanupStrategy int
+
+func (o CleanupStrategy) Option() func(opts *ApplyOptions) {
+	return func(opts *ApplyOptions) {
+		opts.Cleanup = o
+	}
 }
 
+const (
+	// CleanupAlways will trigger a cleanup operation the test context completes.
+	CleanupAlways CleanupStrategy = iota
```

Comment:
Should we just call Option() on all of these as the const (well, `var` technically)? Not sure the value of each caller needing to call .Option?

Reply:
we use the raw one in CleanupStrategy but seems like we could just as easily use the `Cleanup` field? Its only for internal library use as well as far as I can tell so doesn't leak abstractions much.